config/gpt2_data_mix.yaml

```diff
@@ -0,0 +1,34 @@
+data:
+  datasets:
+    - train_urls:
+          - "gs://pubmed-mosaic/openwebtext-sharded/openwebtext_train.{1..128}-of-128.jsonl.gz"
+      validation_urls:
+        - "gs://pubmed-mosaic/openwebtext-sharded/openwebtext_val.{1..8}-of-8.jsonl.gz"
+      cache_dir: "gs://levanter-data/tokenized/openwebtext/"
+      tokenizer: "gpt2"
+      weight: 0.5
+    - id: dlwh/wikitext_103_detokenized
+      cache_dir: "gs://levanter-data/tokenized/wikitext"
+      weight: 0.5
```

Comment:
Yes I do agree that such interface would be nicer for tuning dataset weights. Is there any particular reason that you assign names for each task? 

Maybe an equivalent config for data mixture looks like this:

```
data-mixture:
  data_configs:
    pile:
      train_urls:
        - gs://levanter-data/pile/train/{00..29}.jsonl.zst
      validation_urls:
        - gs://levanter-data/pile/val.jsonl.zst
      cache_dir: "gs://levanter-data/tokenized/pile-old/"
      tokenizer: "EleutherAI/gpt-neox-20b"
    redpajama:
      id: togethercomputer/RedPajama-Data-1T
      cache_dir: gs://levanter-data/tokenized/redpajama/
      tokenizer: EleutherAI/gpt-neox-20b
      splits:
        - train
      rows_per_chunk: 4096
  data_weights:
    pile: 0.6
    redpajama: 0.4
```

I think this could be more flexible and cleaner than the current approach, which is trying to modify the existing `data` field while ensure compatibility (referring back to your comment about LMDatasetConfig).